src/main/java/com/cybozu/kintone/client/connection/Connection.java

```diff
@@ -128,67 +128,45 @@ public JsonElement request(String method, String apiName, String body) throws Ki
         HttpsURLConnection connection = null;
         String response = null;
 
-        boolean isGet = false;
-        if (ConnectionConstants.GET_REQUEST.equals(method)) {
-            isGet = true;
-        }
-
-        URL url = null;
         try {
+            boolean isGet = false;
+
+            if (ConnectionConstants.GET_REQUEST.equals(method)) {
+                isGet = true;
+            }
+    
+            URL url = null;
             url = this.getURL(apiName, null);
-        } catch (MalformedURLException e) {
-            throw new KintoneAPIException("Invalid URL");
-        }
 
-        try {
             connection = openApiConnection(url);
             this.setHTTPHeaders(connection);
             connection.setRequestMethod(method);
-        } catch (Exception e) {
-            throw new KintoneAPIException("can not open connection");
-        }
 
-        connection.setDoOutput(true);
-        connection.setRequestProperty(ConnectionConstants.CONTENT_TYPE_HEADER, JSON_CONTENT);
-        if (isGet) {
-            connection.setRequestProperty(ConnectionConstants.METHOD_OVERRIDE_HEADER, ConnectionConstants.GET_REQUEST);
-        }
+            connection.setDoOutput(true);
+            connection.setRequestProperty(ConnectionConstants.CONTENT_TYPE_HEADER, JSON_CONTENT);
+            if (isGet) {
+                connection.setRequestProperty(ConnectionConstants.METHOD_OVERRIDE_HEADER, ConnectionConstants.GET_REQUEST);
+            }
 
-        try {
             connection.connect();
-        } catch (IOException e) {
-            throw new KintoneAPIException(" cannot connect to host");
-        }
 
-        // send request
-        OutputStream os;
-        try {
+            OutputStream os;
             os = connection.getOutputStream();
-        } catch (IOException e) {
-            throw new KintoneAPIException("an error occurred while sending data");
-        }
-        try {
+
             OutputStreamWriter writer = new OutputStreamWriter(os, StandardCharsets.UTF_8);
             writer.write(body);
             writer.close();
-        } catch (IOException e) {
-            throw new KintoneAPIException("socket error");
-        }
 
-        try {
             checkStatus(connection, body);
             InputStream is = connection.getInputStream();
-            try {
-                response = readStream(is);
-            } finally {
-                is.close();
-            }
-        } catch (Exception e) {
-            throw new KintoneAPIException("an error occurred while receiving data");
-        }
+            response = readStream(is);
 
+            is.close();
 
-        return jsonParser.parse(response);
+            return jsonParser.parse(response);
+        } catch (Exception e) {
+            throw new KintoneAPIException(e.getMessage(), e);
+        }
```

Comment:
OutputStreamWriter will not be closed when exception occur.
cf)
https://docs.oracle.com/javase/9/docs/api/java/lang/AutoCloseable.html
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html